app/controllers/planning_application/immunity_details_controller.rb

```diff
@@ -29,7 +29,7 @@ def edit
 
     def update
       respond_to do |format|
-        if @immunity_detail.update(immunity_details_params)
+        if @immunity_detail.update!(immunity_details_params)
```

Comment:
I think this means we won't reach the `else` block as it'll either raise an error or evaluate as `nil` if nothing is raised (even if the update is successful).

So probs need to rescue any error if we wanted to present the edit page again